spec/lib/journal_formatter/file_link_spec.rb

```diff
@@ -31,64 +31,95 @@
 require 'spec_helper'
 
 RSpec.describe OpenProject::JournalFormatter::FileLink do
-  let(:work_package) { create(:work_package) }
+  let(:work_package) { build(:work_package) }
   let(:journal) { instance_double(Journal, journable: work_package) }
-  let(:file_link) { create(:file_link, container: work_package) }
+  let(:file_link) { create(:file_link, container: work_package, storage: build(:storage)) }
   let(:key) { "file_links_#{file_link.id}" }
 
+  let(:changes) { { "link_name" => file_link.origin_name, "storage_name" => file_link.storage.name } }
+
   subject(:instance) { described_class.new(journal) }
 
   describe '#render' do
-    context 'having the first value being nil, and the second an file link name as string' do
+    context 'having the origin_name as nil' do
+      it 'if the storage name is nil it tries to find it out looking at the file link' do
+        link = "#{Setting.protocol}://#{Setting.host_name}/api/v3/file_links/#{file_link.id}/open"
+
+        changes = { "link_name" => file_link.origin_name, "storage_name" => nil }
```

Comment:
Just  a nit, this can be declared in a let block and re-use it for the other spec as well.
```ruby
let(:changes) { { "link_name" => file_link.origin_name, "storage_name" => nil } }
```

Reply:
Oooh. Good catch!